src/sagemaker/wrangler/ingestion.py

```diff
@@ -0,0 +1,205 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+# language governing permissions and limitations under the License.
+"""Data wrangler helpers for data ingestion."""
+from __future__ import absolute_import
+
+from typing import Dict
+from uuid import uuid4
+from sagemaker.dataset_definition.inputs import (
+    RedshiftDatasetDefinition,
+    AthenaDatasetDefinition,
+)
+
+
+def generate_data_ingestion_flow_from_s3_input(
+    input_name: str,
+    s3_uri: str,
+    s3_content_type: str = "csv",
+    s3_has_header: bool = False,
+    operator_version: str = "0.1",
+    schema: Dict = None,
+):
+    """Generate the data ingestion only flow from s3 input
+
+    Args:
+        input_name (str): the name of the input to flow source node
+        s3_uri (str): uri for the s3 input to flow source node
+        s3_content_type (str): s3 input content type
+        s3_has_header (bool): flag indicating the input has header or not
+        operator_version: (str): the version of the operator
+        schema: (typing.Dict): the schema for the data to be ingested
+    Returns:
+        dict (typing.Dict): A flow only conduct data ingestion with 1-1 mapping
+        output_name (str): The output name used to configure
+        `sagemaker.processing.FeatureStoreOutput`
+    """
+    source_node = {
+        "node_id": str(uuid4()),
+        "type": "SOURCE",
+        "inputs": [],
+        "outputs": [{"name": "default"}],
+        "operator": f"sagemaker.s3_source_{operator_version}",
+        "parameters": {
+            "dataset_definition": {
+                "datasetSourceType": "S3",
+                "name": input_name,
+                "s3ExecutionContext": {
+                    "s3Uri": s3_uri,
+                    "s3ContentType": s3_content_type,
+                    "s3HasHeader": s3_has_header,
+                },
+            }
+        },
+    }
+
+    output_node = {
+        "node_id": str(uuid4()),
+        "type": "TRANSFORM",
+        "operator": f"sagemaker.spark.infer_and_cast_type_{operator_version}",
+        "parameters": {},
+        "inputs": [
+            {"name": "default", "node_id": source_node["node_id"], "output_name": "default"}
+        ],
+        "outputs": [{"name": "default"}],
+    }
+
+    if schema:
+        output_node["trained_parameters"] = schema
+
+    flow = {
+        "metadata": {"version": 1, "disable_limits": False},
+        "nodes": [source_node, output_node],
+    }
+
+    return flow, f'{output_node["node_id"]}.default'
```

Comment:
Let's extract 66-85 to a private function across those three variants?

Reply:
sure will do.

Reply:
the sdk integration test can leverage the DW processor Ethan checked in, we probably will also have one in our service integ test suite. If we did it, our canary will run it.

for canary test, I wonder if sdk has its own canary tests?